modular/gater/admin_handler.go

```diff
@@ -377,6 +378,169 @@ func (g *GateModular) getChallengeInfoHandler(w http.ResponseWriter, r *http.Req
 	log.CtxDebug(reqCtx.Context(), "succeed to get challenge info")
 }
 
+// getChallengeInfoV2Handler handles get challenge piece info request. Current only greenfield
+// validator can challenge piece is store correctly. The challenge piece info includes:
+// the challenged piece data, all piece hashes and the integrity hash. The challenger
+// can verify the info whether are correct by comparing with the greenfield info.
+// The difference between the v2 interface and the old interface is the format of the returned results,
```

Comment:
Maybe `the old interface(v1)` is better.

Reply:
Fixed.